File: zenscript-code-model/src/main/java/raylras/zen/model/Preprocessor.java
package raylras.zen.model;

import java.util.Arrays;

public record Preprocessor(String header, String[] data) {

    public static Preprocessor create(String raw) {
        String[] split = raw.split(" ");
        String header = split[0];
        String[] data = Arrays.copyOfRange(split, 1, split.length);
        return new Preprocessor(header, data);
    }

}

File: zenscript-code-model/src/main/kotlin/raylras/zen/model/Preprocessor.kt
package raylras.zen.model

data class Preprocessor(val raw: String) {
    val split: List<String> = raw.substring(1).split(" ".toRegex())

    val name: String
        get() = split[0]

    val rest: List<String>
        get() = split.subList(1, split.size)
}
